ccl/sqlproxyccl/tenant: TestResume failed

ccl/sqlproxyccl/tenant.TestResume [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9442771?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9442771?buildTab=artifacts#/) on master @ [ff2c4a3454a63bd74fe9a2feed53a7bdcfaf5191](https://github.com/cockroachdb/cockroach/commits/ff2c4a3454a63bd74fe9a2feed53a7bdcfaf5191):


```
=== RUN   TestResume
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f215c8f665df7837d22d3ffe15cbecee/logTestResume410630612
    directory_cache_test.go:511: starting tenant 40
    directory_cache_test.go:267: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:267
        	Error:      	Condition never satisfied
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = context canceled
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f215c8f665df7837d22d3ffe15cbecee/logTestResume410630612
--- FAIL: TestResume (19.86s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-sessions @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestResume.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-26553

ccl/sqlproxyccl/tenant.TestResume [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9482800?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9482800?buildTab=artifacts#/) on master @ [8e6f530644eb35313a479af529b1f672c12dca38](https://github.com/cockroachdb/cockroach/commits/8e6f530644eb35313a479af529b1f672c12dca38):


```
=== RUN   TestResume
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f215c8f665df7837d22d3ffe15cbecee/logTestResume1676103822
    directory_cache_test.go:511: starting tenant 40
    directory_cache_test.go:267: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:267
        	Error:      	Condition never satisfied
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = context canceled
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f215c8f665df7837d22d3ffe15cbecee/logTestResume1676103822
--- FAIL: TestResume (20.31s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestResume.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/sqlproxyccl/tenant.TestResume [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9516858?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9516858?buildTab=artifacts#/) on master @ [85e41ca8d3d9edacf5ee3061a2591b159b9b0502](https://github.com/cockroachdb/cockroach/commits/85e41ca8d3d9edacf5ee3061a2591b159b9b0502):


```
=== RUN   TestResume
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f215c8f665df7837d22d3ffe15cbecee/logTestResume1337181087
    directory_cache_test.go:511: starting tenant 40
    directory_cache_test.go:267: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:267
        	Error:      	Condition never satisfied
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = context canceled
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:259: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:259
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f215c8f665df7837d22d3ffe15cbecee/logTestResume1337181087
--- FAIL: TestResume (20.18s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100981 ccl/sqlproxyccl/tenant: TestResume failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestResume.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/sqlproxyccl/tenant.TestResume [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9761613?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9761613?buildTab=artifacts#/) on master @ [1f3419e178bdba544f74d9c9e14a4682efd18028](https://github.com/cockroachdb/cockroach/commits/1f3419e178bdba544f74d9c9e14a4682efd18028):


```
=== RUN   TestResume
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f215c8f665df7837d22d3ffe15cbecee/logTestResume1284402600
    directory_cache_test.go:507: starting tenant 40
    directory_cache_test.go:263: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:263
        	Error:      	Condition never satisfied
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = context canceled
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f215c8f665df7837d22d3ffe15cbecee/logTestResume1284402600
--- FAIL: TestResume (20.27s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101429 ccl/sqlproxyccl/tenant: TestResume failed [C-test-failure O-robot branch-release-23.1.0]
- #100981 ccl/sqlproxyccl/tenant: TestResume failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestResume.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/sqlproxyccl/tenant.TestResume [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9881138?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9881138?buildTab=artifacts#/) on master @ [7d02e061759599a2b054ee06c71524443f77d20a](https://github.com/cockroachdb/cockroach/commits/7d02e061759599a2b054ee06c71524443f77d20a):


```
=== RUN   TestResume
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f215c8f665df7837d22d3ffe15cbecee/logTestResume2121985308
    directory_cache_test.go:507: starting tenant 40
    directory_cache_test.go:263: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:263
        	Error:      	Condition never satisfied
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = context canceled
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f215c8f665df7837d22d3ffe15cbecee/logTestResume2121985308
--- FAIL: TestResume (18.83s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101429 ccl/sqlproxyccl/tenant: TestResume failed [C-test-failure O-robot branch-release-23.1.0]
- #100981 ccl/sqlproxyccl/tenant: TestResume failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestResume.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/sqlproxyccl/tenant.TestResume [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9905473?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9905473?buildTab=artifacts#/) on master @ [ed08739c22ecd246c9745d0f301976023c0d004b](https://github.com/cockroachdb/cockroach/commits/ed08739c22ecd246c9745d0f301976023c0d004b):


```
=== RUN   TestResume
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f215c8f665df7837d22d3ffe15cbecee/logTestResume1329633720
    directory_cache_test.go:507: starting tenant 40
    directory_cache_test.go:263: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:263
        	Error:      	Condition never satisfied
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = context canceled
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    directory_cache_test.go:255: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/tenant_test/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go:255
        	            				src/runtime/asm_amd64.s:1594
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestResume
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f215c8f665df7837d22d3ffe15cbecee/logTestResume1329633720
--- FAIL: TestResume (19.69s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101429 ccl/sqlproxyccl/tenant: TestResume failed [C-test-failure O-robot branch-release-23.1.0]
- #100981 ccl/sqlproxyccl/tenant: TestResume failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestResume.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go
+++ b/pkg/ccl/sqlproxyccl/tenant/directory_cache_test.go
@@ -411,7 +411,6 @@ func TestCancelLookups(t *testing.T) {
 func TestResume(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.ScopeWithoutShowLogs(t).Close(t)
-	skip.UnderDeadlockWithIssue(t, 71365)
 
 	tenantID := roachpb.MustMakeTenantID(40)
 	const lookupCount = 5
@@ -437,12 +436,15 @@ func TestResume(t *testing.T) {
 		}(i)
 	}
 
-	var processes map[net.Addr]*tenantdirsvr.Process
 	// Eventually the tenant process will be resumed.
-	require.Eventually(t, func() bool {
+	var processes map[net.Addr]*tenantdirsvr.Process
+	testutils.SucceedsSoon(t, func() error {
 		processes = tds.Get(tenantID)
-		return len(processes) == 1
-	}, 10*time.Second, 100*time.Millisecond)
+		if len(processes) != 1 {
+			return errors.Newf("expected 1 processes found %d", len(processes))
+		}
+		return nil
+	})
 
 	// Wait until background goroutines complete.
 	wait.Wait()

